multicast/multicast.go

```diff
@@ -0,0 +1,285 @@
+	if err != nil {
+		return err
+	}
+
+	for _, ciliumPod := range ciliumPods {
+		//Print cilium pod info
+		_, err := m.printCiliumPodInfo(ciliumPod)
+		if err != nil {
+			return err
+		}
+
+		//List multicast subscribers
+		cmd := []string{"cilium-dbg", "bpf", "multicast", "subscriber", "list", target}
+		output, err := m.execInCiliumPod(ctx, ciliumPod, cmd)
+		if err != nil {
+			fmt.Fprintf(m.params.Writer, "\n")
+			continue
+		}
+		fmt.Fprintln(m.params.Writer, output)
+	}
+
+	return nil
+}
+
+// AddAllNodes add CiliumInternalIPs of all nodes to the specified multicast group as subscribers in every cilium-agent
+func (m *Multicast) AddAllNodes() error {
+	if m.params.MulticastGroupIP == "" {
+		return fmt.Errorf("group-ip must be specified")
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), m.params.WaitDuration)
+	defer cancel()
+	ciliumPods, err := m.listCiliumPods(ctx)
+	if err != nil {
+		return err
+	}
+
+	//Create a map of ciliumInternalIPs of all nodes
+	ipToPodMap := make(map[v2.NodeAddress]string)
+	for _, ciliumPod := range ciliumPods {
+		ciliumInternalIP, err := m.getCiliumInternalIP(ciliumPod.Spec.NodeName)
+		if err != nil {
+			return err
+		}
+		ipToPodMap[ciliumInternalIP] = ciliumPod.Name
+	}
+
+	for _, ciliumPod := range ciliumPods {
+		//Print cilium pod info
+		_, err := m.printCiliumPodInfo(ciliumPod)
+		if err != nil {
+			return err
+		}
+
+		//If there are not specified multicast group, create it
+		cmd := []string{"cilium-dbg", "bpf", "multicast", "subscriber", "list", m.params.MulticastGroupIP}
+		_, err = m.execInCiliumPod(ctx, ciliumPod, cmd)
+		if err != nil {
+			cmd = []string{"cilium-dbg", "bpf", "multicast", "group", "add", m.params.MulticastGroupIP}
+			_, err := m.execInCiliumPod(ctx, ciliumPod, cmd)
+			if err != nil {
+				fmt.Fprintf(m.params.Writer, "Unable to create multicast group %s in %s\n", m.params.MulticastGroupIP, ciliumPod.Name)
+				continue
+			}
+		}
+
+		//Add all ciliumInternalIPs of all nodes to the multicast group as subscribers
+		cnt := 0
+		for ip, podName := range ipToPodMap {
+			if ip.IP != "" && ciliumPod.Name != podName {
+				cmd = []string{"cilium-dbg", "bpf", "multicast", "subscriber", "add", m.params.MulticastGroupIP, ip.IP}
+				_, err := m.execInCiliumPod(ctx, ciliumPod, cmd)
+				if err == nil {
+					cnt++
+				}
+			}
+		}
+		fmt.Fprintf(m.params.Writer, "Added %d subscribers to multicast group %s\n\n", cnt, m.params.MulticastGroupIP)
+	}
+
+	return nil
+}
+
+// DelAllNodes delete CiliumInternalIPs of all nodes from the specified multicast group's subscribers in every cilium-agent
+func (m *Multicast) DelAllNodes() error {
+	if m.params.MulticastGroupIP == "" {
+		return fmt.Errorf("group-ip must be specified")
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), m.params.WaitDuration)
+	defer cancel()
+	ciliumPods, err := m.listCiliumPods(ctx)
+	if err != nil {
+		return err
+	}
+
+	//Create a map of ciliumInternalIPs of all nodes
```

Comment:
Consider creating a helper method for this; the same pattern is used in the add and del methods.